pkg/adapter/v2/main_injection_test.go

```diff
@@ -61,28 +61,32 @@ func TestMain_WithController_DisableHA(t *testing.T) {
 	})
 
 	env := ConstructEnvOrDie(func() EnvConfigAccessor { return &myEnvConfig{} })
-	MainWithInformers(ctx,
-		"mycomponent",
-		env,
-		func(ctx context.Context, processed EnvConfigAccessor, client cloudevents.Client) Adapter {
-			env := processed.(*myEnvConfig)
-
-			if env.Mode != "mymode" {
-				t.Error("Expected mode mymode, got:", env.Mode)
-			}
-
-			if env.Sink != "http://sink" {
-				t.Error("Expected sinkURI http://sink, got:", env.Sink)
-			}
-
-			if leaderelection.HasLeaderElection(ctx) {
-				t.Error("Expected no leader election, but got leader election")
-			}
-			return &myAdapter{}
-		})
+	done := make(chan bool)
```

Comment:
why do this in a test?

Reply:
It's to make sure MainWithInformers runs before view.Unregister.